invenio_rdm_pure/source/rdm/run/synchronizer.py

```diff
@@ -81,25 +98,79 @@ def synchronize_research_outputs(
         """
         while True:
             research_outputs = get_research_outputs(
-                pure_api_key,
-                pure_api_url,
+                self.pure_api_key,
+                self.pure_api_url,
                 size,
                 offset,
             )  # Fetch research outputs from Pure
             if research_outputs:
                 break
             else:
                 time.sleep(0.001)
-        # Check if list is not empty
         converter = Converter()
         for research_output in research_outputs:
             try:
-                converter.convert_pure_json_to_marc21_xml(research_output)
+                record_xml = converter.convert_pure_json_to_marc21_xml(research_output)
+                if Marc21Record.is_valid_marc21_xml_string(record_xml):
+                    # Store record with the help of marc21 module
+                    with app.app_context():
+                        record = self.create_record(record_xml)
+                        files = self.download_record_files(research_output)
+                        self.attach_files_to_record(files, record)
+                        self.delete_record_files(files)
             except RuntimeError:
                 traceback.print_exc()
 
-        # TODO: Store record with the help of marc21 module
-        pass
+    def create_record(self, record_xml: str) -> RecordItem:
+        """Create Invenio record from Marc21XML string."""
+        identity = Identity(self.pure_user_id)
+        identity.provides.add(any_user)
```

Comment:
i am not sure if it is wise to add any_user to the identity. but i have also no suggestions, so go ahead with it.